Update to Testcontainers 2.0.2

gsmet opened the pull request:
@gastaldi this includes all your work + a commit to work around the Strimzi issue. I got surprised that Debezium had it working and I gave it a try.

Note that there are probably things that wouldn't work but from my testing our use cases seems to somehow work. I did test only one Kafka module that uses the Strimzi container though.

Let's see how it goes.

dependabot[bot] committed:
Bumps `testcontainers.version` from 1.21.3 to 2.0.2.

Updates `org.testcontainers:testcontainers-bom` from 1.21.3 to 2.0.2
- [Release notes](https://github.com/testcontainers/testcontainers-java/releases)
- [Changelog](https://github.com/testcontainers/testcontainers-java/blob/main/CHANGELOG.md)
- [Commits](https://github.com/testcontainers/testcontainers-java/compare/1.21.3...2.0.2)

Updates `org.testcontainers:testcontainers` from 1.21.3 to 2.0.2
- [Release notes](https://github.com/testcontainers/testcontainers-java/releases)
- [Changelog](https://github.com/testcontainers/testcontainers-java/blob/main/CHANGELOG.md)
- [Commits](https://github.com/testcontainers/testcontainers-java/compare/1.21.3...2.0.2)

---
updated-dependencies:
- dependency-name: org.testcontainers:testcontainers-bom
  dependency-version: 2.0.2
  dependency-type: direct:production
  update-type: version-update:semver-major
- dependency-name: org.testcontainers:testcontainers
  dependency-version: 2.0.2
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Co-authored-by: George Gastaldi <gegastaldi@gmail.com>
Co-authored-by: Guillaume Smet <guillaume.smet@gmail.com>
Signed-off-by: dependabot[bot] <support@github.com>

gsmet committed:
Work around strimzi-test-containers not being compatible with TC 2.x

There are efforts going on here:
https://github.com/dajudge/kindcontainer/pull/402
https://github.com/strimzi/test-container/issues/166

But for now, it seems good enough to add a few exclusions.

gastaldi commented on bom/application/pom.xml:
```diff
@@ -196,7 +196,7 @@
+        <testcontainers.version>2.0.2</testcontainers.version> <!-- Make sure to also update docker-java.version to match its needs -->
         <docker-java.version>3.4.2</docker-java.version> <!-- must be the version Testcontainers use: https://central.sonatype.com/artifact/org.testcontainers/testcontainers -->
         <!-- Check the compatibility matrix (https://github.com/opensearch-project/opensearch-testcontainers) before upgrading: -->
         <opensearch-testcontainers.version>2.0.2</opensearch-testcontainers.version>
```
And this: 

```suggestion
        <opensearch-testcontainers.version>4.0.0</opensearch-testcontainers.version>
```

gsmet committed:
Update OpenSearch Testcontainers support to 2.1.3

For now, we can't update to 4.x as it only supports Java 21+.

I created
https://github.com/opensearch-project/opensearch-testcontainers/issues/313
to see if they could support Java 17 again.

gsmet committed:
Switch OpenSearch tests to use Testcontainers

This way, we can check they are still working correctly.

holly-cummins committed:
Remove test workarounds for test containers 1

gsmet commented:
This should be ready now.

rsvoboda commented:
I think this should have breaking change label and a note in https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.31

Motivation: Testcontainers 2 introduce change in GAVs and some packages

geoand commented:
> I think this should have breaking change label

It's definitely a breaking, Quarkus LangChain4j for example fails on with this change. Adding the label as a result
